ddtrace/internal/forksafe.py

```diff
@@ -55,3 +57,55 @@ def _after_fork():
         ddtrace_after_in_child()
 
     threading._after_fork = _after_fork  # type: ignore[attr-defined]
+
+
+_resetable_objects = weakref.WeakSet()  # type: weakref.WeakSet[ResetObject]
+
+
+def _reset_objects():
+    # type: (...) -> None
+    for obj in _resetable_objects:
+        try:
+            obj._reset_object()
+        except Exception:
+            log.exception("Exception ignored in object reset forksafe hook %r", obj)
+
+
+register(_reset_objects)
+
+
+class ResetObject(wrapt.ObjectProxy):
+    """An object wrapper object that is fork-safe and resets itself after a fork.
```

Comment:
It's not really clear what "resets" means here. Might be worth saying "reinstantiate" to be direct?